### ISSUE TITLE
503 Server Unavailable.

### ISSUE THREAD

**Issue by auszone** [sunspot/sunspot#683]
I'm a beginner of sunspot, and I followed the tutorial.
1. add 'sunspot_rails' & 'sunspot_solr' into my gemfile & bundle install
2. rails generate sunspot_rails:install
3. and start the solr server via sunspot::solr::start
And when I tried to save an object into database or reindex my data, it showed the error:
```
RSolr::Error::Http: RSolr::Error::Http - 503 Service Unavailable
Error:     Server is shutting down
```
Why is that?
Does anyone else encounter this problem too?

**Comment by IslamAzab** [comment]
How does your `config/sunspot.ym` file look like?

If you are using the default values, then can you open this `localhost:8982/solr`?

**Comment by auszone** [comment]
I think I'm using the default values, since I didn't edit it at all. And when I open this `localhost:8982/solr` it give me the same response.
```
HTTP ERROR 503

Problem accessing /solr/. Reason:

    Server is shutting down
```

**Comment by IslamAzab** [comment]
Can you run `bundle exec rake sunspot:solr:run --trace` and check why solr is shutting down?

**Comment by auszone** [comment]
it says NO JSP Support for /solr, did not find org.apache.jasper.servlet.JspServlet
why is that? 
```
** Invoke sunspot:solr:run (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute sunspot:solr:run
2015-04-16 18:16:26.390:INFO:oejs.Server:jetty-8.1.8.v20121106
2015-04-16 18:16:26.412:INFO:oejdp.ScanningAppProvider:Deployment monitor /usr/local/Cellar/ruby/2.2.1/lib/ruby/gems/2.2.0/gems/sunspot_solr-2.2.0/solr/contexts at interval 0
2015-04-16 18:16:26.420:INFO:oejd.DeploymentManager:Deployable added: /usr/local/Cellar/ruby/2.2.1/lib/ruby/gems/2.2.0/gems/sunspot_solr-2.2.0/solr/contexts/solr.xml
2015-04-16 18:16:26.488:INFO:oejw.WebInfConfiguration:Extract jar:file:/usr/local/Cellar/ruby/2.2.1/lib/ruby/gems/2.2.0/gems/sunspot_solr-2.2.0/solr/webapps/solr.war!/ to /private/var/folders/yf/lvh3tgkd69sdvl4xdj06mr8h0000gn/T/jetty-0.0.0.0-8982-solr.war-_solr-any-/webapp
2015-04-16 18:16:27.851:INFO:oejw.StandardDescriptorProcessor:NO JSP Support for /solr, did not find org.apache.jasper.servlet.JspServlet
Null identity service, trying login service: null
Finding identity service: null
2015-04-16 18:16:27.873:INFO:oejsh.ContextHandler:started o.e.j.w.WebAppContext{/solr,file:/private/var/folders/yf/lvh3tgkd69sdvl4xdj06mr8h0000gn/T/jetty-0.0.0.0-8982-solr.war-_solr-any-/webapp/},/usr/local/Cellar/ruby/2.2.1/lib/ruby/gems/2.2.0/gems/sunspot_solr-2.2.0/solr/webapps/solr.war
2015-04-16 18:16:27.873:INFO:oejsh.ContextHandler:started o.e.j.w.WebAppContext{/solr,file:/private/var/folders/yf/lvh3tgkd69sdvl4xdj06mr8h0000gn/T/jetty-0.0.0.0-8982-solr.war-_solr-any-/webapp/},/usr/local/Cellar/ruby/2.2.1/lib/ruby/gems/2.2.0/gems/sunspot_solr-2.2.0/solr/webapps/solr.war
2015-04-16 18:16:28.034:INFO:oejs.AbstractConnector:Started SocketConnector@0.0.0.0:8982
```

**Comment by IslamAzab** [comment]
What do you have in your `solr/data` folder? You might not be able to access your data folder or it has corrupted data, or you have a typo in your xml config file. These are the possible reasons I've found so far.

**Comment by auszone** [comment]
oh i just solved it, it's because my $JAVA_HOME is not set correctly.

**Comment by masterkain** [comment]
I'm on yosemite with the same issue, care to share what your JAVA_HOME looks like?

mine is `/Library/Java/JavaVirtualMachines/jdk1.8.0_20.jdk/Contents/Home`

**Comment by masterkain** [comment]
solved downgrading java to 1.7

**Comment by auszone** [comment]
$JAVA_HOME: `/Library/Java/JavaVirtualMachines/jdk1.8.0_45.jdk/Contents/Home`
FYI, my java version is 1.8.
